### DISCUSSION TITLE
How do I install Ninja as generator in Github Actions

### DISCUSSION THREAD

**Discussion by mscofield0** [TheLartians/Greeter#146] [Q&A] [answered]
I would like to specify the generator directly, but the workflow does not have Ninja installed currently. How would I modify my workflow setup to install Ninja?

**Comment by ClausKlein** [comment, marked as answer]
`pip install ninja`

**Reply by mscofield0** [reply]
Where do I specify that in the Github workflow?

**Reply by mscofield0** [reply]
And what is this `**/cpm_modules` that appears in the workflow files?

**Reply by ClausKlein** [reply]
you may see a example here: https://github.com/ClausKlein/minijson_reader/blob/30e0f41aea7d72ef5a7a72701a1ec4b81bd28323/.github/workflows/cmake.yml#L32

**Reply by mscofield0** [reply]
Thank you Claus, this really helped me.

**Comment by ClausKlein** [comment]
`CPM_SOURCE_CACHE: ${{ github.workspace }}/cpm_modules`

the cloned git repos are here, with this cache module, they are restored ...

see https://github.com/TheLartians/ModernCppStarter/runs/3972841226?check_suite_focus=true
